cpp/src/groupby/hash/groupby.cu

```diff
@@ -539,7 +539,7 @@ std::unique_ptr<table> groupby_null_templated(table_view const& keys,
 
   // Extract the populated indices from the hash map and create a gather map.
   // Gathering using this map from sparse results will give dense results.
-  rmm::device_vector<size_type> gather_map;
+  rmm::device_uvector<size_type> gather_map(0, stream);
   size_type map_size;
   std::tie(gather_map, map_size) = extract_populated_keys(*map, keys.num_rows(), stream);
```

Comment:
This creates a zero-size uvector. To avoid ITM, perhaps it would be better to skip the std::tie until we have structured bindings. e.g. 

```suggestion
  auto map_and_size = extract_populated_keys(*map, keys.num_rows(), stream);
  rmm::device_uvector<size_type> gather_map{map_and_size.first};
  size_type const map_size = map_and_size.second;
```

Reply:
Make sure you're not copying. 
```
rmm::device_uvector<size_type> gather_map{std::move(map_and_size.first)};
```

Reply:
Updated. (Awaiting C++17 - structure binding declaration.)